app/api/models/statistics.py

```diff
@@ -113,6 +113,8 @@ class GenomeStatistics(BaseModel):
     def __init__(self, **data):
         data["_compiled_data"] = {
             stats_item["name"]: stats_item["statisticValue"]
+            if stats_item["statisticValue"] != "null"
```

Comment:
@kamaldodiya In python it's recommended if your one line loop isn't one line anymore it's better to use the complete for structure:
for i in list:
   print(i)

Reply:
> In python it's recommended 
Don't mind changing it but May I know why it is recommended or why it shouldn't be this way ?

Reply:
1. It's because of code readability that it's  recommended try to avoid complex loop structures.
2. Here we can just avoid assigning "null" values entirely which will help with the memory management and better use of python dictionaries.
3. Something like without else statement:
`   def __init__(self, **data):
        data["_compiled_data"] = {}
        for stat_item in data["_raw_data"]:
            if stat_item["statisticValue"] != "null":
                data["_compiled_data"].update({stat_item["name"]: stat_item["statisticValue"]})`

Reply:
> It's because of code readability that it's recommended try to avoid complex loop structures.

Dictionary (list) comprehension are more readable than for loop albeit, as you mentioned, for simple structure.

> Here we can just avoid assigning "null" values entirely which will help with the memory management and better use of python dictionaries.

I think this is good idea. Pydantic will assign None to the missing fields right ?

```
data["_compiled_data"] = {
    si["name"]: si["statisticValue"]
    for si in data["_raw_data"]
    if si["statisticValue"] != "null"
}
```

https://google.github.io/styleguide/pyguide.html#27-comprehensions--generator-expressions 
See Section 2.7.4 (Yes) Example 2
```
result = [{'key': value} for value in iterable
        if a_long_filter_expression(value)]
```

Reply:
It does state in the cons that it's against readability in the link that you provided and it does indicate that the statement should fit in one line.
and it's a recommended PEP8 style for when the line is getting longer
https://peps.python.org/pep-0008/#whitespace-in-expressions-and-statements

Also in Effective python book, Item 2, expressions and statements references to Zen of  Python and PEP8 . it recommends  if, for, while and except to be spread into multi lines for clarity.(Although my personal preference is still to use  it when it's a simple statement!)
> There should be one-- and preferably only one --obvious way to do it. Although that way may not be obvious at first unless you're Dutch.


> I think this is good idea. Pydantic will assign None to the missing fields right ?

Yes, That's right. And for our code it's going to be something like:

```python
        data["_compiled_data"] = {
            stats_item["name"]: stats_item["statisticValue"]
            for stats_item in data["_raw_data"]
            if stats_item["statisticValue"] != "null"
        }
```

Reply:
> It does state in the cons that it's against readability

Complicated comprehensions or generator expressions can be hard to read. Here it's just for and if condition checking for non null. I don't think its complicated

> it does indicate that the statement should fit in one line.

 Each portion must fit on one line: mapping expression, for clause, filter expression. 
We have Dictionary creation in one line, for in second and if in third. All less than 80 chracters :)

Reply:
> Complicated comprehensions or generator expressions can be hard to read.
In this case it's just for and if condition checking for non null. I don't think its complicated

It's usually used in way more simple use-cases.
 
Even in the first version it didn't fit in one line and it needed to be split in separate lines. What is the point of having separate lines for each statement when using single line expressions? 

At the end there's one rule which is "Make it Work"! :D